x-pack/platform/plugins/shared/onechat/server/services/tools/tool_types/esql/create_schema.ts

```diff
@@ -41,7 +41,13 @@ export function createSchemaFromParams(params: EsqlToolConfig['params']): z.ZodO
     }
 
     if (param.optional) {
-      field = field.optional();
+      if (param.defaultValue !== undefined) {
+        // Use default value instead of making it optional
+        field = field.default(param.defaultValue);
+      } else {
+        // No default value, make it optional
+        field = field.optional();
+      }
```

Comment:
Totally optional because we didn't have any in the first place - but if you feel like it, this schema generation function could benefit from a few unit tests for sure.

Reply:
https://github.com/elastic/kibana/pull/238008/commits/8d87349bd64c73a87d1ddc9fc55af830f23da52f